examples/utils.h

```diff
@@ -128,39 +141,41 @@ static void* malloc_orDie(size_t size)
 }
 
 /*! loadFile_orDie() :
- * Read size bytes from a file.
- * 
+ * Read size bytes from a file. If buffer is not provided (i.e., buffer == null),
+ * malloc will be called to allocate one.
+ *
  * Note: This function will send an error to stderr and exit if it
  * cannot read data from the given file path.
- * 
+ *
  * @return If successful this function will return a pointer to read
  * data otherwise it will printout an error to stderr and exit.
  */
-static void* loadFile_orDie(const char* fileName, size_t* size)
+static void* loadFile_orDie(const char* fileName, size_t* size, void* buffer, int bufferSize)
```

Comment:
This works. All paths look properly checked.

However, in term of _interface design_,
I suspect it's not great to have a function which sometimes allocate a buffer and sometimes doesn't (depending on its arguments), even if it's clearly documented.

I feel rather in favor of 2 versions,
one which just load data into an existing buffer,
and another one which creates a buffer and uses the first version to load it.
The second version should have a name which makes it very clear that an allocation is happening, since it means a `free` operation will have to be triggered later on.

cc @terrelln for additional point of view.

Reply:
Totally agree! I feel a bit hesitant to do that too...

Reply:
code refactored.